src/test/spec/unified_runner/operation.rs

```diff
@@ -1644,6 +1649,123 @@ impl TestOperation for RunCommand {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(super) struct RunCursorCommand {
+    command: Document,
+    // We don't need to use this field, but it needs to be included during deserialization so that
+    // we can use the deny_unknown_fields tag.
+    #[serde(rename = "commandName")]
+    _command_name: String,
+
+    #[serde(flatten)]
+    options: RunCursorCommandOptions,
+    session: Option<String>,
+}
+
+impl TestOperation for RunCursorCommand {
+    fn execute_entity_operation<'a>(
+        &'a self,
+        id: &'a str,
+        test_runner: &'a TestRunner,
+    ) -> BoxFuture<'a, Result<Option<Entity>>> {
+        async move {
+            let command = self.command.clone();
+            let db = test_runner.get_database(id).await;
+
+            let options = RunCursorCommandOptions::builder()
+                .max_time(self.options.max_time)
+                .cursor_type(self.options.cursor_type)
+                .selection_criteria(self.options.selection_criteria.clone())
+                .comment(self.options.comment.clone())
+                .batch_size(self.options.batch_size)
+                .build();
+
+            let result = match &self.session {
+                Some(session_id) => {
+                    with_mut_session!(test_runner, session_id, |session| async {
+                        let mut cursor = db
+                            .run_cursor_command_with_session(command, options, session)
+                            .await?;
+                        cursor.stream(session).try_collect::<Vec<_>>().await
+                    })
+                    .await?
+                }
+                None => {
+                    let cursor = db.run_cursor_command(command, options).await?;
+                    cursor.try_collect::<Vec<_>>().await?
+                }
+            };
+
+            Ok(Some(bson::to_bson(&result)?.into()))
+        }
+        .boxed()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct CreateCommandCursor {
+    command: Document,
+    // We don't need to use this field, but it needs to be included during deserialization so that
+    // we can use the deny_unknown_fields tag.
+    #[serde(rename = "commandName")]
+    _command_name: String,
+
+    #[serde(flatten)]
+    options: RunCursorCommandOptions,
+    session: Option<String>,
+}
+
+impl TestOperation for CreateCommandCursor {
+    fn execute_entity_operation<'a>(
+        &'a self,
+        id: &'a str,
+        test_runner: &'a TestRunner,
+    ) -> BoxFuture<'a, Result<Option<Entity>>> {
+        async move {
+            let command = self.command.clone();
+            let db = test_runner.get_database(id).await;
+
+            let options = RunCursorCommandOptions::builder()
```

Comment:
https://github.com/mongodb/mongo-rust-driver/pull/922#discussion_r1291560936

(copying this from the previous PR, also applies on line 1676 above)

Reply:
Done.